### ISSUE TITLE
idea: split out budo's guts

### ISSUE THREAD

**Issue by yoshuawuyts** [mattdesl/budo#122]
So I've been working on a bunch of streaming providers for `{html,css,js}`, and I was wondering what's missing before it _could_ (not saying it should) be used as the heart of `budo`?

I'm currently including `tiny-lr` for livereloading out of the box (https://github.com/yoshuawuyts/bankai/pull/3), and the only difference with `budo` would be that the glob LR params can't be changed. Would that matter?

Because `bankai` uses streams everywhere this would also help further #60, which I still think would be pretty great if we could pull off. Probably also simplifies `budo`'s internals a bit, which might be nice?

Lmk what you think / how `bankai` can improve 😁Thanks!

**Comment by mattdesl** [comment]
Yeah, I have split some features out already (see `watchify-middleware`), and ideally I'd like to split out a few more things.

It would be hard to use a module that isn't somewhat purpose-built for budo. There are a lot of subtle things that have grown into budo over time; things like manually handling a debounce for faster incremental bundles, reporting installed watchify version, handling chokidar edge cases, handling babelify and ansi-colored build errors, etc.
